contrib/fuzz/metadata_fuzz_test.go

```diff
@@ -0,0 +1,428 @@
+   You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package fuzz
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	bolt "go.etcd.io/bbolt"
+
+	"github.com/containerd/containerd/v2/core/containers"
+	"github.com/containerd/containerd/v2/core/content"
+	"github.com/containerd/containerd/v2/core/images"
+	"github.com/containerd/containerd/v2/core/leases"
+	"github.com/containerd/containerd/v2/core/metadata"
+	"github.com/containerd/containerd/v2/core/snapshots"
+	"github.com/containerd/containerd/v2/pkg/namespaces"
+	"github.com/containerd/containerd/v2/plugins/content/local"
+	"github.com/containerd/containerd/v2/plugins/snapshots/native"
+)
+
+func testEnv(t *testing.T) (context.Context, *bolt.DB, func(), error) {
+	dirname := t.TempDir()
+	db, err := bolt.Open(filepath.Join(dirname, "meta.db"), 0644, nil)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	ctx = namespaces.WithNamespace(ctx, "testing")
+	return ctx, db, func() {
+		db.Close()
+		cancel()
+	}, nil
+}
+
+func FuzzImageStore(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		imageStoreOptions := map[int]string{
+			0: "Create",
+			1: "List",
+			2: "Update",
+			3: "Delete",
+		}
+
+		ctx, db, cancel, err := testEnv(t)
+		if err != nil {
+			return
+		}
+		defer cancel()
+		store := metadata.NewImageStore(metadata.NewDB(db, nil, nil))
+		f := fuzz.NewConsumer(data)
+		noOfOperations, err := f.GetInt()
+		if err != nil {
+			return
+		}
+		maxOperations := 50
+		for i := 0; i < noOfOperations%maxOperations; i++ {
+			opType, err := f.GetInt()
+			if err != nil {
+				return
+			}
+			switch imageStoreOptions[opType%len(imageStoreOptions)] {
+			case "Create":
+				i := images.Image{}
+				err := f.GenerateStruct(&i)
+				if err != nil {
+					return
+				}
+				_, _ = store.Create(ctx, i)
+			case "List":
+				newFs, err := f.GetString()
+				if err != nil {
+					return
+				}
+				_, _ = store.List(ctx, newFs)
+			case "Update":
+				i := images.Image{}
+				err := f.GenerateStruct(&i)
+				if err != nil {
+					return
+				}
+				_, _ = store.Update(ctx, i)
+			case "Delete":
+				name, err := f.GetString()
+				if err != nil {
+					return
+				}
+				_ = store.Delete(ctx, name)
+			}
+		}
+	})
+}
+
+func FuzzLeaseManager(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		leaseManagerOptions := map[int]string{
```

Comment:
(multiple occurrences) Seems like this could be a `[]string`?  And maybe the values should be `const` here so they can be defined in one place instead of two (lines 113-118 and 139-188)?

If this was migrated code, we can do that improvement in a separate commit.

Reply:
> (multiple occurrences) Seems like this could be a []string? And maybe the values should be const here.

yeah agreed.

This is some existing code, so updated in a separate commit that move strings to consts and change maps to []string.